src/terrain/TerrainHeightHillsCraters.cpp

```diff
@@ -24,21 +24,27 @@ TerrainHeightFractal<TerrainHeightHillsCraters>::TerrainHeightFractal(const Syst
 }
 
 template <>
-double TerrainHeightFractal<TerrainHeightHillsCraters>::GetHeight(const vector3d &p) const
+void TerrainHeightFractal<TerrainHeightHillsCraters>::GetHeights(const std::vector<vector3d> &vP, std::vector<double> &heightsOut) const
 {
-	double continents = octavenoise(GetFracDef(0), 0.5, p) - m_sealevel;
-	if (continents < 0) return 0;
-	// == TERRAIN_HILLS_NORMAL except river_octavenoise
-	double n = 0.3 * continents;
-	double distrib = river_octavenoise(GetFracDef(2), 0.5, p);
-	double m = GetFracDef(1).amplitude * river_octavenoise(GetFracDef(1), 0.5 * distrib, p);
-	// cliffs at shore
-	if (continents < 0.001)
-		n += m * continents * 1000.0f;
-	else
-		n += m;
-	n += crater_function(GetFracDef(3), p);
-	n += crater_function(GetFracDef(4), p);
-	n *= m_maxHeight;
-	return (n > 0.0 ? n : 0.0);
+	for (size_t i = 0; i < vP.size(); i++) {
+		const vector3d &p = vP[i];
+		const double continents = octavenoise(GetFracDef(0), 0.5, p) - m_sealevel;
+		if (continents < 0.0)
+			heightsOut.at(i) = 0.0;
+
+		// == TERRAIN_HILLS_NORMAL except river_octavenoise
+		double n = 0.3 * continents;
+		double distrib = river_octavenoise(GetFracDef(2), 0.5, p);
+		double m = GetFracDef(1).amplitude * river_octavenoise(GetFracDef(1), 0.5 * distrib, p);
```

Comment:
While the compiler can indeed (somewhat) trivially see that the underlying `m_fracdef` is const during the loop and move this load outside of the loop in release mode, I'd ideally prefer `GetFracDef(n)` be replaced with `m_fracdef[n]` for debug-mode performance, given that we call it up to 8 times per loop iteration.

Reply:
If it's ok I'm not going to touch this one for this PR. It's just used everywhere in the terrain noise and always has been. So I'd rather do that as a separate PR

Reply:
That's fine - I was mentioning it as it'd be a simple find-replace of `GetFracDef` -> `m_fracDef`. A separate PR would be nice though!